substrate/frame/collective/src/lib.rs

```diff
@@ -882,32 +920,20 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		hash: &T::Hash,
 		length_bound: u32,
 		weight_bound: Weight,
-	) -> Result<(<T as Config<I>>::Proposal, usize), DispatchError> {
-		let key = ProposalOf::<T, I>::hashed_key_for(hash);
-		// read the length of the proposal storage entry directly
-		let proposal_len =
-			storage::read(&key, &mut [0; 0], 0).ok_or(Error::<T, I>::ProposalMissing)?;
+	) -> Result<(<T as Config<I>>::Proposal, u32), DispatchError> {
+		let encoded =
+			T::Preimages::fetch(&hash, None).map_err(|_| Error::<T, I>::ProposalMissing)?;
+		let proposal_len = encoded.len() as u32;
 		ensure!(proposal_len <= length_bound, Error::<T, I>::WrongProposalLength);
-		let proposal = ProposalOf::<T, I>::get(hash).ok_or(Error::<T, I>::ProposalMissing)?;
+		let proposal = <T as Config<I>>::Proposal::decode(&mut &encoded[..])
+			.map_err(|_| Error::<T, I>::DecodeFailed)?;
 		let proposal_weight = proposal.get_dispatch_info().weight;
 		ensure!(proposal_weight.all_lte(weight_bound), Error::<T, I>::WrongProposalWeight);
-		Ok((proposal, proposal_len as usize))
+		Ok((proposal, proposal_len))
 	}
 
-	/// Weight:
-	/// If `approved`:
-	/// - the weight of `proposal` preimage.
-	/// - two events deposited.
-	/// - two removals, one mutation.
-	/// - computation and i/o `O(P + L)` where:
-	///   - `P` is number of active proposals,
-	///   - `L` is the encoded length of `proposal` preimage.
-	///
-	/// If not `approved`:
-	/// - one event deposited.
-	/// Two removals, one mutation.
-	/// Computation and i/o `O(P)` where:
-	/// - `P` is number of active proposals
```

Comment:
Why has this been removed?

Reply:
It is a legacy comment from ages ago, that refers to a function that was the precursor of `do_approve_proposal` and `do_disapprove_proposal`.
[source](https://github.com/paritytech/polkadot-sdk/commit/0573f1408d4a179a1eb0b4ae95a8a95a289d158f#diff-c71493c1b073034aebb47c3a7961fd47975174582fa76db48b967037682ad054R337-R376)